### ISSUE TITLE
Npc's and My character enter infinite loop

### ISSUE THREAD

**Issue by x7GoSu** [CleverRaven/Cataclysm-DDA#62485]
### Describe the bug

Npcs and player character will  look for items endlessly. 

### Steps to reproduce

1. Set up a construction zone or vehicle deconstruction zone.
2. don't have the items
3. watch character go back and forth looking for something that isn't there.

### Expected behavior

For a check that there isn't the required tool items whatever command ends.

### Screenshots

![infinloop](https://user-images.githubusercontent.com/20464973/204708522-f798a02c-986a-4243-be3a-c4b737a0526b.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.2251 (22H2)
- Game Version: 10a3f17 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Blaze Industries - UndeadPeople Tileset [BlazeIndustriesUDP],
    No Hope [no_hope],
    Dark Days of the Dead [classic_zombies],
    CustomContent [CustomContent],
    Add Rice [rice],
    Locked and Loaded [Locked_and_Loaded],
    No Rail Stations [No_Rail_Stations],
    Tamable Wildlife [Tamable_Wildlife]
]

### Additional context
[TheWalkingDead.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/10119665/TheWalkingDead.zip)

If anyone knows how to stop that activity or edit the save to stop him would appreciate it.  Just spent 3 hours grinding prof lol

**Comment by andrei8l** [comment]
Upload your save file please

**Comment by x7GoSu** [comment]
I cannot it is too large tip?

**Comment by andrei8l** [comment]
Delete the cache and .mm1 folders and try again. If it's still too large, use a third-party website like https://send.vis.ee/

**Comment by x7GoSu** [comment]
Alright thank you done.

**Comment by andrei8l** [comment]
/confirmed caused by broken NPC exclusion check

<details>
<summary>Possible fix</summary>

```diff
diff --git a/src/activity_item_handling.cpp b/src/activity_item_handling.cpp
index 345e91138a..34a3e981df 100644
--- a/src/activity_item_handling.cpp
+++ b/src/activity_item_handling.cpp
@@ -1119,7 +1119,7 @@ static activity_reason_info can_do_activity_there( const activity_id &act, Chara
                     already_working_indexes.push_back( guy.activity_vehicle_part_index );
                 }
             }
-            if( player_character.activity_vehicle_part_index != -1 ) {
+            if( !you.is_avatar() && player_character.activity_vehicle_part_index != -1 ) {
                 already_working_indexes.push_back( player_character.activity_vehicle_part_index );
             }
         }
```
There are many other bugs here though and zone activity code needs to be heavily reworked or rewritten from scratch.
</details>

I don't think we'll address this before stable so I've also manually fixed your save by cancelling the activity: [Walking Dead.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/10120426/Walking.Dead.zip)

**Comment by Maleclypse** [comment]
Closed by #69379
